api/dao/containerstorage.py

```diff
@@ -429,3 +435,162 @@ def get_all_for_targets(self, target_type, target_ids,
         if collection_id:
             query['collections'] = collection_id
         return self.get_all_el(query, user, projection)
+
+
+class AnalysisStorage(ContainerStorage):
+    def __init__(self):
+        super(AnalysisStorage, self).__init__('analyses', use_object_id=True)
+
+
+    def get_parent(self, parent_type, parent_id):
+        parent_storage = ContainerStorage.factory(parent_type)
+        return parent_storage.get_container(parent_id)
+
+
+    def get_analyses(self, parent_type, parent_id, inflate_job_info=False):
+        parent_type = containerutil.singularize(parent_type)
+        parent_id = bson.objectid.ObjectId(parent_id)
+        analyses = self.get_all_el({'parent.type': parent_type, 'parent.id': parent_id}, None, None)
+        if inflate_job_info:
+            for analysis in analyses:
+                self.inflate_job_info(analysis)
+        return analyses
+
+
+    def get_fileinfo(self, _id, filename=None):
+        analysis = self.get_container(_id)
+        files = analysis.get('files')
+        if files is None:
+            return None
+        if filename:
+            for f in files:
+                if f.get('name') == filename:
+                    return [f]
+        else:
+            return files
+
+
+    def fill_values(self, analysis, cont_name, cid, origin):
+        parent = self.get_parent(cont_name, cid)
+        defaults = {
+            'parent': {
+                'type': containerutil.singularize(cont_name),
+                'id': bson.objectid.ObjectId(cid)
+            },
+            '_id': bson.objectid.ObjectId(),
+            'created': datetime.datetime.utcnow(),
+            'modified': datetime.datetime.utcnow(),
+            'user': origin.get('id'),
+            'permissions': parent['permissions'],
+            'public': parent.get('public', False),
+        }
+        for key in defaults:
```

Comment:
I'm curious about why you chose to use `setdefault` rather than creating a default and then calling `return default.update(analysis)`? Just a style choice? I just very rarely use `setdefault()`.